doc_src/index.hdr.in

```diff
@@ -17,7 +17,199 @@
+<tr><td>switch and data as arguments &rarr;<td>&rarr; exit code
+</table>
+
+This leads to another way of programming and especially of combining commands:
+
+There are two ways to combine shell commands:
+
+- Commands can pass on their streams to each other: one command takes the output stream of another command as its input stream and the two commands execute in parallel. This is called piping, see <a href="#piping">Piping</a>.
+
+Example:
+
+\fish
+# Every line of the `ls` command is immediatelly passed on to the `grep` command
+>ls -l | grep "my topic"
+\endfish
+
+- Commands can pass on all their output as a chunk</b>: the output stream of one command is bundled and taken as data argument for the second command. This is called command substitution, see <a href="#expand-command-substitution">Command Substitution</a>.
+
+Example:
+\fish
+# the output of the inner `ls` command is taken as the input argument for the outer `echo` command
+>echo (ls a*)
+\endfish
+
+\subsection shebang-line Shebang Line
+
+Since script for shell commands can be written in many different languages, they need to carry information about what interpreter is needed to execute them: For this they are expected to have a first line, the shebang line, which names an executable for this purpose:
+
+Example: 
+
+A scripts written in `bash` it would need a first line like this:
+
+```
+#!/bin/bash
+```
+
+- this line tells the shell to execute the file with the *bash* interpreter, that is located at the path `/bin/bash`.
+
+For a script, written in another language, just replace the interpreter `/bin/bash` with the language interpreter of that other language (for example `/bin/python` for a `python` script)
+
+This line is only needed when scripts are executed by another interpreter, so for *fish* internal commands, that are executed by *fish* the shebang line is not necessary.
+
+<a href="#shell"><-</a><hr>
+
+
+\section install Installation and Start
+
+This section is on how to install, uninstall, start and exit a *fish* shell and on how to make *fish* the default shell:
+
+- <a href="#installation">Install</a>: How to install *fish*
+- <a href="#default-shell">Default Shell</a>: How to switch to *fish* as the default shell
+- <a href="#start">Starting and Exiting</a>: How to start and exit a *fish* shell
+- <a href="#uninstall">Uninstall</a>: How to uninstall *fish*
+- <a href="#bash">Executing Bash</a>: How to execute *bash* commands in *fish*
+
+\subsection installation Installation
+
+Instructions for installing fish are on the <a href="https://fishshell.com/">fish homepage</a>. Search that page for "Go fish". 
+
+To install the development version of *fish* see the instructions at the <a href="https://github.com/fish-shell/fish-shell">project's GitHub page</a>.
+
+\subsection default-shell Default Shell
+
+You can make *fish* your default shell by adding *fish*'s  executable in two places:
+- add `/usr/local/bin/fish`  to  `/etc/shells`
```

Comment:
Perhaps introduce this with "Your operating system will generally start a particular shell by default. This may be changed in your terminal program, or you can set it for all terminals by using the `chsh` command."

Most packages add the line to `/etc/shells`, and it needs to be done by the system administrator, so could you add something describing how to check if it's there already and noting that the the use of the root or administrative account might be required?

Reply:
Would this be better:

"Your operating system will generally start a particular shell by default. This may be changed in your terminal program or on the command-line with the `chsh` command.

In any case you need to know the excutable of your fish shell: you can get this using fish in the following way:

```
>fish # set the current shell to fish
yourusername@yourcomputer ~>type fish          
/usr/local/bin/fish 
```

How to adjust the preferences for your terminal applications will depend on the type of your computer and its operating system and eventually the type of your terminal application.  But there is also a way to adjust this setting on the command-line, that will be described here:

You can use the `chsh` command to do this. 
This command takes:
- a `-s` option to enter the executeable file for the shell
- a `-u` option to change the setting for a specific user: without this option the setting will be changed for all users using the computer

```
>fish # set the current shell to fish
yourusername@yourcomputer ~> chsh -s /usr/local/bin/fish  -u yourusername   
```

You may get an error message, doing this: 

```
chsh: /usr/local/bin/fish: non-standard shell
```

This means that that your computer does not know about the fish shell yet. It has to be added to `/etc/shells` on your computer. In order to do this you will need administration rights."

Is this better now? What about the tutorial? I think if I write it this detailed the changing of the default shell should not be in both the tutorial and the documentation, but rather in the documentation and linked to by the tutorial. What do you think on that?

Reply:
Yes, that's better - I would put a link in the tutorial perhaps.